### ISSUE TITLE
在bge-m3上用全量bge-m3-data数据集做post training，发现效果会显著变差是为什么？

### ISSUE THREAD

**Issue by syusukexrj** [FlagOpen/FlagEmbedding#1422]
/ssd2/bge-m3
煤炉 春游 0.3848
红领巾 大肚子 0.417
红领巾 少先队 0.501

epoch1：/ssd2/checkpoint-6133
煤炉 春游 0.503
红领巾 大肚子 0.537
红领巾 少先队 0.6226

epoch2：/ssd2/checkpoint-12264
煤炉 春游 0.4958
红领巾 大肚子 0.5376
红领巾 少先队 0.617

可以发现词的区分度明显变差。下面是我的训练参数

`export WANDB_MODE=disabled

train_data="\
    /ssd2/bge-m3-data"

output_dir="/ssd2/"
# set large epochs and small batch size for testing
num_train_epochs=2
per_device_train_batch_size=8

# set num_gpus to 2 for testing
num_gpus=8

model_args="\
    --model_name_or_path /ssd2/bge-m3 \
"

data_args="\
    --train_data $train_data \
    --train_group_size 8 \
    --query_max_len 512 \
    --passage_max_len 512 \
    --pad_to_multiple_of 8 \
    --knowledge_distillation False \
"

training_args="\
    --output_dir $output_dir \
    --overwrite_output_dir \
    --learning_rate 3e-6 \
    --bf16 \
    --num_train_epochs $num_train_epochs \
    --per_device_train_batch_size $per_device_train_batch_size \
    --gradient_accumulation_steps 4 \
    --dataloader_drop_last True \
    --warmup_ratio 0.1 \
    --gradient_checkpointing \
    --deepspeed ./ds_stage1.json \
    --weight_decay 0.01 \
    --logging_steps 5 \
    --report_to tensorboard \
    --save_strategy epoch \
    --temperature 0.02 \
    --sentence_pooling_method cls \
    --normalize_embeddings True \
    --kd_loss_type m3_kd_loss \
    --unified_finetuning True \
    --use_self_distill True \
    --fix_encoder False \
    --self_distill_start_step 0 \
"

cmd="torchrun --nproc_per_node $num_gpus \
    -m FlagEmbedding.finetune.embedder.encoder_only.m3 \
    $model_args \
    $data_args \
    $training_args \
"

echo $cmd
eval $cmd`

**Comment by 545999961** [comment]
1. 检索器微调过程中是会用到 in-batch negative 的，所以`gradient_accumulation_steps`这个参数并没有实际增大 in-batch negative 数量，对于对比学习训练意义是不大的，需要增加`per_device_train_batch_size`来扩大实际的 batch size
2. 针对特定的任务可以用特定任务的数据集进行训练，bge-m3本身已经用bge-m3-data训练过了，想进一步增强领域内能力可以用领域内数据进行微调
